api/src/main/java/io/strimzi/api/kafka/model/AutoRestart.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.DescriptionFile;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Representation of the AutoRestart configuration.
+ */
+@DescriptionFile
+@Buildable(
+    editableEnabled = false,
+    builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public class AutoRestart implements UnknownPropertyPreserving, Serializable {
+    private static final long serialVersionUID = 1L;
+    private boolean enabled = false;
```

Comment:
Wasn't this set to `true` originally? I would do this in one of the following ways:
* Default to `true` here. But leaving the `autoRestart` field in `AbstractConnectorSpec` default to null (as you already have it). That way, it will be disabled by default (as autoRestart is not set by default). But it would allow to just set it to `autoRestart: {}` to enable the auto-restart.
* Or I would simply make it required without any default.

I would do one of the above because it seems a bit weird to me that `autoRestart: {}` would actually mean disabled auto-restart. Seems counterintuitive.